src/games/strategy/engine/data/Unit.java

```diff
@@ -37,6 +38,10 @@ public UnitType getUnitType() {
   public PlayerID getOwner() {
     return m_owner;
   }
+  
+  public UnitAttachment getUnitAttachment() {
+  	return (UnitAttachment) m_type.getAttachment("unitAttachment");
```

Comment:
This is good, information hiding, the clients of this class do not have to know about quite as many details. Changes like this add up quickly to allow a person to use high level of abstractions

:+1: